File: src/main/java/org/example/chessplatformbe/controller/dto/response/StreamResponseDTO.java
package org.example.chessplatformbe.controller.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class StreamResponseDTO {
    private Integer id;

    private String name;

    private LocalDateTime creationDateTime;

    private String streamUrl;

    private Integer streamerId;
}

File: src/controller/dto/response/stream_response_dto.rs
pub struct StreamResponseDTO {
    id: u32,
    name: String,
    creation_date_time: Date,
    stream_url: String,
    streamer_id: u32,
}

impl StreamResponseDTO {
    pub fn new(
        id: u32,
        name: String,
        creation_date_time: Date,
        stream_url: String,
        streamer_id: u32,
    ) -> Self {
        Self {
            id,
            name,
            creation_date_time,
            stream_url,
            streamer_id,
        }
    }
}
